packages/vite/src/node/env.ts

```diff
@@ -26,12 +27,10 @@ export function loadEnv(
 
   const parsed = Object.fromEntries(
     envFiles.flatMap((file) => {
-      const path = lookupFile(envDir, [file], {
-        pathOnly: true,
-        rootDir: envDir,
-      })
-      if (!path) return []
-      return Object.entries(parse(fs.readFileSync(path)))
+      const filePath = path.join(envDir, file)
+      if (!tryStatSync(filePath)?.isFile()) return []
+
+      return Object.entries(parse(fs.readFileSync(filePath)))
```

Comment:
We were passing `rootDir: envDir` here, so only level would be checked by `lookupFile` but the call makes it seems env dirs are being looked up in the fs. Replaced it with a single `tryStatSync`